dotnet/src/Microsoft.AutoGen/Agents/AgentBase.cs

```diff
@@ -108,7 +108,16 @@ await this.InvokeWithActivityAsync(
                 break;
         }
     }
-
+    protected async Task Store(AgentState state)
+    {
+        await _context.Store(state).ConfigureAwait(false);
+        return;
+    }
+    protected async Task<T> Read<T>(AgentId agentId) where T : IMessage, new()
```

Comment:
is not clear this is reading state since T is constrained on Imessage, the Store side is more explicit (because of variable name)

Reply:
I think this should be constrained on AgentState instead. Watch for a new PR.

Reply:
READY